cryptofeed/exchange/probit.py

```diff
@@ -0,0 +1,191 @@
+                "quantity":"0"
+            }]
+        }
+        '''
+        pair = pair_exchange_to_std(msg['market_id'])
+
+        is_snapshot = msg.get('reset', False)
+
+        if is_snapshot:
+            self.l2_book[pair] = {ASK: sd(), BID: sd()}
+
+            for entry in msg["order_books"]:
+                price = Decimal(entry['price'])
+                quantity = Decimal(entry['quantity'])
+                side = BID if entry['side'] == "buy" else ASK
+                self.l2_book[pair][side][price] = quantity
+
+            await self.book_callback(self.l2_book[pair], L2_BOOK, pair, True, None, timestamp, timestamp)
+        else:
+            delta = {BID: [], ASK: []}
+
+            for entry in msg["order_books"]:
+                price = Decimal(entry['price'])
+                quantity = Decimal(entry['quantity'])
+                side = BID if entry['side'] == "buy" else ASK
+                if quantity == 0:
+                    if price in self.l2_book[pair][side]:
+                        del self.l2_book[pair][side][price]
+                    delta[side].append((price, 0))
+                else:
+                    self.l2_book[pair][side][price] = quantity
+                    delta[side].append((price, quantity))
+
+            await self.book_callback(self.l2_book[pair], L2_BOOK, pair, False, delta, timestamp, timestamp)
+
+    async def message_handler(self, msg: str, timestamp: float):
+        msg = json.loads(msg, parse_float=Decimal)
+
+        # Probit can send multiple type updates in one message so we avoid the use of elif
+        if 'recent_trades' in msg:
+            await self._trades(msg, timestamp)
+        if 'order_books' in msg:
+            await self._l2_update(msg, timestamp)
+        # Probit has a 'ticker' channel, but it provide OHLC-last data, not BBO px.
+
+    async def subscribe(self, websocket):
+        self.__reset()
+
+        if self.config:
+            for chan in self.config:
+                for pair in self.config[chan]:
+                    await websocket.send(json.dumps({"type": "subscribe",
+                                                     "channel": "marketdata",
+                                                     "filter": [chan],
+                                                     "interval": 100,
+                                                     "market_id": pair,
+                                                     }))
+        else:
+            for pair in self.pairs:
+                await websocket.send(json.dumps({"type": "subscribe",
+                                                 "channel": "marketdata",
+                                                 "filter": list(self.channels),
+                                                 "interval": 100,
+                                                 "market_id": pair,
+                                                 }))
+                await asyncio.sleep(0.3)
```

Comment:
is this needed? Unless you need to sync data from a REST endpoint or the like I'm guessing this is not needed

Reply:
you're right, the `await asyncio.sleep(0.3)` is probably not needed. Removing.